ssl/statem/statem_lib.c

```diff
@@ -1981,12 +1981,13 @@ int ssl_version_supported(const SSL_CONNECTION *s, int version,
          ++vent) {
         if (vent->cmeth != NULL
                 && ssl_version_cmp(s, version, vent->version) == 0
-                && ssl_method_error(s, vent->cmeth()) == 0
+                && ssl_method_error(s, s->server ? vent->smeth()
```

Comment:
Should this be also conditioned on s->server?

Reply:
Not sure I understand the question? It *is* conditioned on `s->server`?

Reply:
And them I would just factor out this ternary to a temporary SSL_METHOD * variable.